lib/Config/Project.php

```diff
@@ -223,4 +234,46 @@ public function get_pot_path() : string {
 		}
 		return "{$path}.pot";
 	}
+
+	/**
+	 * Parse filename in config and replace requested arguments with their values.
+	 *
+	 * @since 0.0.0
+	 *
+	 * @param  string $filename
+	 * @param  array  $arg_values
+	 * @param  string $locale
+	 * @return string
+	 * @throws FilenameArgumentNotAvailable
+	 */
+	private function parse_filename(
+		string $filename,
+		array $arg_values,
+		string $locale = ''
+	) : string {
+
+		// Replace locale
+		$filename = str_replace( '{$locale}', $locale, $filename );
+
+		$matches = [];
+		preg_match_all( '/\{\$([a-zA-Z0-9_-]*)\}/', $filename, $matches );
+
+		foreach ( $matches[1] as $string_arg ) {
+
+			if ( ! isset( $arg_values[ $string_arg ] ) ) {
```

Comment:
Two more tests should be performed:

- Ensure what comes out is a string.
- Blacklist or whitelist some values. Exemple, what happens if one parameter is placeholder is {filename}?

Reply:
The matches is a `preg_match_all` result, which is always of type string because it's a search on a string.

Reply:
String replace always returns a string, unless the $replace is an array which would give an Array conversion error.

If a placeholders value is something like {filename} there's no problem since we only run the regex once before replacing, so whatever the value is, it will be placed there.